### PR TITLE
fix: improve logo sizing behavior with sticky header

### LINKED ISSUE #1561: Sticky Header: Centred logos can get cut off
It looks like the styles for the sticky header may be causing centred logos to get cut off at some breakpoints. 

Needs further investigating/testing.

### PR DESCRIPTION
**Pull request by laurelfulford**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-theme/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR is to fix some issues introduced with #1563, a PR meant to stop the centred logo from getting cut off when the sticky header was enabled. That PR ended up causing the logo to become distorted, or display too large in some cases. 

Closes #1561.

### How to test the changes in this Pull Request:

I think the best way to test this PR is by comparing it to how the header looked prior to #1563 getting merged -- the main two things to check for are making sure the centred logo does not get cut off with this PR applied, and also that the logo size is the same as it was prior to #1563.

1. On your test site, navigate to Customize > Header Settings > Appearance and enable the sticky header, and centre the logo.
2. Navigate to Customize > Header Settings > Subpage Header, and enable the simplified subpage header. 
3. Upload a horizontal logo, and make it fairly small using the logo sizer (~200px wide). 
4. View the regular and non-AMP versions of the page; note the size of the logo -- it will be larger than in the Customizer, and the non-AMP version may be distorted (depending on the logo's aspect ratio) -- this it the main issue we're trying to fix: 

Customizer: 

image

Front-end: 

image

5. Open two tabs with the master branch, on of the homepage, and one of a subpage.

Homepage:

image

Subpage:
image

6. Check out the pre-sticky-header fix version of the theme (`git checkout 1.0.0-alpha.33-340-g36c52397`), and run `npm run build`.
7. Open two more tabs, and open the homepage and a subpage in each -- these will be left open for comparison.

Homepage: 
image

Subpage: 
image

8. Check out this PR and run `npm run build.`
9. Open two more tabs, one with the homepage and one with a subpage. 

Homepage: 
image

Subpage: 
image

10. Compare the tabs; note that the ones running `master` have a larger, maybe distorted version of the logo.
11. Compare the `1.0.0-alpha.33-340-g36c52397` tabs and the one from this PR; confirm that the logos are the same size. 
12. Switch to a horizontal logo, and make it very large.
13. Repeat the steps to compare the pre-1563 commit (`1.0.0-alpha.33-340-g36c52397`) to this PR and confirm the logos are sized the same

`1.0.0-alpha.33-340-g36c52397`
image

image

This PR:
image

http://build.newspack.test/2021/10/article-summary-test/

14. With this PR, make the browser window narrower -- until it switches to the mobile  header -- and confirm that the logo doesn't get cut off: 

image

image

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Review by adekbadek — APPROVED** [review]
I find it a bit hard to follow the testing instructions –
1. "the non-AMP version may be distorted (depending on the logo's aspect ratio)" - what aspect ratio would make the logo distorted? An image to use could be added to the PR description. 
2. Why is testing against pre-sticky-header version required? I'd expect an issue to be reproducible on `master` and fixed on the PR branch.

I've also observed that the subpage header custom logo is not used below ~780px screen width (the regular logo is), but maybe that's expected.

That said, I did not notice any issues with the code and changing the use of `height` & `max-width` to using only the latter looks like it would fix any cropping issues.

**Comment by laurelfulford** [comment]
Thanks Adam! 

> what aspect ratio would make the logo distorted? An image to use could be added to the PR description.

🤦‍♀️ D'oh! I had intended to include one, then didn't circle back to it -- sorry about that! It's not so helpful now, but one specific case I ran into was with the Newspack logo in the 'subpage short header', so the aspect ratio was roughly 3:1: 

image

> Why is testing against pre-sticky-header version required?

This was mostly to make sure the logo size was close to what it was prior to the changes in master -- the distortion issues could have been fixed, but the logo sizes still could have been larger than they were originally (larger logos were a specific complaint that came up from the last update). 

> I've also observed that the subpage header custom logo is not used below ~780px screen width (the regular logo is), but maybe that's expected.

This is expected! The alternative logo is used in case you need a lighter version of the logo to contrast against the 'featured image behind' or 'featured image beside' styles -- in both cases, when you have the simplified subpage header turned on, the header doesn't have a solid background colour on larger screen sizes. It 'comes back' on smaller screen sizes.

**Comment by yogeshbeniwal** [comment]
@laurelfulford Sticky header is till resulting in stretching of logo on sub header. Highlighting if more such reports come across. Keeping header non sticky for the time being.
